PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = 'account.fiscal.position'
-    _inherit = 'pos.config'
-    _inherit = 'pos.order'
-    _inherit = 'pos.payment'
-    _inherit = 'pos.session'
-    _inherit = 'res.config.settings'

Total Changes: 19

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/pos_restaurant/models/account_fiscal_position.py
+++ b/addons/pos_restaurant/models/account_fiscal_position.py
@@ -3,7 +3,7 @@ from odoo.osv.expression import OR
 
 
 class AccountFiscalPosition(models.Model):
-    _inherit = 'account.fiscal.position'
+    _inherit = ['account.fiscal.position']
 
     @api.model
     def _load_pos_data_domain(self, data):

--- a/addons/pos_restaurant/models/pos_config.py
+++ b/addons/pos_restaurant/models/pos_config.py
@@ -8,7 +8,7 @@ from odoo.tools import convert
 
 
 class PosConfig(models.Model):
-    _inherit = 'pos.config'
+    _inherit = ['pos.config']
 
     iface_splitbill = fields.Boolean(string='Bill Splitting', help='Enables Bill Splitting in the Point of Sale.')
     iface_printbill = fields.Boolean(string='Bill Printing', help='Allows to print the Bill before payment.')

--- a/addons/pos_restaurant/models/pos_order.py
+++ b/addons/pos_restaurant/models/pos_order.py
@@ -2,8 +2,9 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 from odoo import api, fields, models
 
+
 class PosOrder(models.Model):
-    _inherit = 'pos.order'
+    _inherit = ['pos.order']
 
     table_id = fields.Many2one('restaurant.table', string='Table', help='The table where this order was served', index='btree_not_null', readonly=True)
     customer_count = fields.Integer(string='Guests', help='The amount of customers that have been served by this order.', readonly=True)

--- a/addons/pos_restaurant/models/pos_payment.py
+++ b/addons/pos_restaurant/models/pos_payment.py
@@ -4,8 +4,8 @@
 from odoo import api, fields, models
 
 
-class PosConfig(models.Model):
-    _inherit = 'pos.payment'
+class PosPayment(models.Model):
+    _inherit = ['pos.payment']
 
     def _update_payment_line_for_tip(self, tip_amount):
         """Inherit this method to perform reauthorization or capture on electronic payment."""

--- a/addons/pos_restaurant/models/pos_restaurant.py
+++ b/addons/pos_restaurant/models/pos_restaurant.py
@@ -7,7 +7,6 @@ from odoo.exceptions import UserError
 
 class RestaurantFloor(models.Model):
 
-    _name = 'restaurant.floor'
     _description = 'Restaurant Floor'
     _order = "sequence, name"
     _inherit = ['pos.load.mixin']
@@ -81,9 +80,9 @@ class RestaurantFloor(models.Model):
 
         return True
 
+
 class RestaurantTable(models.Model):
 
-    _name = 'restaurant.table'
     _description = 'Restaurant Table'
     _inherit = ['pos.load.mixin']
 

--- a/addons/pos_restaurant/models/pos_session.py
+++ b/addons/pos_restaurant/models/pos_session.py
@@ -4,8 +4,9 @@
 from odoo import models, api
 import json
 
+
 class PosSession(models.Model):
-    _inherit = 'pos.session'
+    _inherit = ['pos.session']
 
     @api.model
     def _load_pos_data_models(self, config_id):

--- a/addons/pos_restaurant/models/res_config_settings.py
+++ b/addons/pos_restaurant/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     pos_floor_ids = fields.Many2many(related='pos_config_id.floor_ids', readonly=False)
     pos_iface_printbill = fields.Boolean(compute='_compute_pos_module_pos_restaurant', store=True, readonly=False)
